src/search/srde.c

```
@@ -0,0 +1,546 @@
+  const int max_iters = (cfg->max_iters > 0) ? cfg->max_iters : MAX_ITER_DE;
+  const int max_evals = (cfg->max_evals > 0) ? cfg->max_evals : MAX_EVAL_DE;
+  real weight = (cfg->weight > 0.0) ? cfg->weight : (real)0.8;
+  real cr = (cfg->crossover > 0.0) ? cfg->crossover : (real)0.9;
+  if (cr > (real)1.0) {
+    cr = (real)1.0;
+  }
+  const real span = (cfg->init_span > 0.0) ? cfg->init_span : (real)1.0;
+
+  /* Initialize RNG */
+  sr_rng rng;
+  const uint64_t seed = (cfg->seed > 0) ? cfg->seed : 1ULL;
```

Clang (14.0.6)

[warning] variable 'seed' is not initialized

(cppcoreguidelines-init-variables)